Kyle

sweetstyle11 opened the pull request.

Commit:
Kyle

Commit:
first response works

Commit:
adding text to alerts

Commit:
trying to get more questions

ghost commented:
Is there a way to get the boolean data from the correct column on the database into my javascript file? I'm wondering because I'm trying to get the rest of the questions to appear, and I'm not sure how to loop through the questions in the database. I created a "params" variable and I can pass that number to my controller so my first thought would be just to increment that value, but I'm not sure how to get the correct answer. Right now I've just hardcoded the correct response but that only works for the first question.

halfghaninne commented:
Hm. Did you decide not to use the `correct_answer` class method pre-defined in the Questions model?

ghost commented:
Yes, I didn't think it was possible to call that method using javascript?

ghost commented:
I guess I'm confused how to send information from javascript back into the sinatra files. I can get the basic text to show up in my alert messages but I don't know how to send the users input back. So all the true or false checking is happening in javascript, and I don't feel like thats necessarily a great approach?

Commit:
three questions show up

halfghaninne commented:
> I didn't think it was possible to call that method using javascript?

You're right it's not! That work is done on the server-side of things.

>  I can get the basic text to show up in my alert messages 

How have you been sending basic text back to your alert messages? Where is that text coming from?

Commit:
question

ghost commented:
I'm really struggling to get the information that the user inputs from the prompt to go back to the server. Do I need to make another XMLHttpRequest? if so, how do I only send that request if the user has submitted the prompt?

sumeetjain commented:
Alright, your code is very close–but I can see where you took a wrong turn.
1. There is no `click` event for `prompt`. Like `alert`, `prompt` halts execution. When the user types something in and submits, then execution continues. So `userResponse.addEventListener("click",function(){...` is not needed.
2. Your `answer` XHR object should be defined _inside_ of `request.addEventListener("load",...`, because–in terms of the sequence of events–you want to submit a request to your server for information about that after the `prompt` has retrieved the user's answer. And that happens inside of the `request.addEventListener("load",...` block.

Commit:
returns correct answer

Commit:
checks for correct answers

Commit:
phase 1 complete I think

Commit:
possible answers show

Commit:
one question works

Commit:
multiple questions work

Commit:
Keeps score acurately but....
when the game is over for some reason the score is doubled and then some weird addition happens resulting in the final score message returning not the right answer.

Commit:
score is not acurate

ghost commented:
I've got a question about how my game is keeping score. I believe it has something to do with what we covered in lecture this morning but I'm struggling to get the bottom of it. I have a conditional that should only increment the score if the response text is "That is correct.", and if I never answer any questions correct the score stays 0. Thats good. But, If I answer more than one question correctly the score appears to increment twice. Why would that happen if the conditional for incrementing score is supposed to be waiting from a 'load' event when the server returns correct or not?

sumeetjain commented:
Have you put a debugger inside the block for `if(userResponse.responseText == "That is correct."){` to see if 'correctness' is happening when it shouldn't? Did that reveal anything? (It's okay if it didn't–I'm just sharing my own debugging process for this. I haven't solved the issue.)

ghost commented:
That helped. It appears that the first time the page loads. The block checks for correctness once, and everything is good. But after I click "next question" if the answer is correct, that block actually run more than once. Now I'm starting to think that the problem is because of the way I structorred my script to repeat after I click next. I was unsure about this from the beginning. Is there a better way to display the questions before I click next?

sumeetjain commented:
The problem is that `var sendAnswer = new XMLHttpRequest();` is defined outside of `submitAnswer.addEventListener("click", function(){...`.

As a global variable, it exists outside of any one event. So every time someone clicks `submitAnswer`, they are running the code that adds a 'load' event listener to `sendAnswer`. Those event listeners stack on top of each other.

Moving `var sendAnswer = new XMLHttpRequest();` inside the `submitAnswer.addEventListener("click", function(){...` block fixes this issue, because it's a brand new XHR being made–so it doesn't have the event listener from the previous click of `submitAnswer`.
